onnx/defs/nn/defs.cc

```diff
@@ -2190,6 +2190,198 @@ ONNX_OPERATOR_SET_SCHEMA(
         }
         )ONNX"));
 
+void col2imShapeInference(InferenceContext& ctx) {
+  propagateElemTypeFromInputToOutput(ctx, 0, 0);
+
+  // All inputs shapes are required
+  if (!hasNInputShapes(ctx, 3)) {
+    return;
+  }
+
+  // TODO: Assume image_shape has correct spatial dimensions for next validations
+  //       An alternative is get the the number of spatial dimensions as an input
+  if (ctx.getInputType(1)->tensor_type().shape().dim_size() != 1) {
+    fail_shape_inference("image_shape tensor must have rank 1.");
+  }
+  size_t n_input_dims = ctx.getInputType(1)->tensor_type().shape().dim(0).dim_value();
```

Comment:
The dim could be symbolic or unknown here. I suggest replacing this line with
```cc
Dim n_input_dims ;
unifyInputDim (ctx, 1, 0, n_input_dims);
```
and update the uses as commented below.

Reply:
done